### ISSUE TITLE
Table ContextMenu Selection should be unselected after clicking away

### ISSUE THREAD

**Issue by davisb10** [primefaces/primeng#7821]
If you have a PrimeNG PRO Support subscription please post your issue at;

https://pro.primefaces.org 

where our team will respond within 4 business hours.

If you do not have a PrimeNG PRO Support subscription, fill-in the report below. Please note that
your issue will be added to the waiting list of community issues and will be reviewed on a first-come first-serve basis, as a result, the support team is unable to guarantee a specific schedule on when it will be reviewed. Thank you for your understanding.

Current Queue Time for Review
Without PRO Support: ~8-12 weeks.
With PRO Support: 1 hour

**I'm submitting a ...**  (check one with "x")
```
[X] bug report => Search github for a similar issue or PR before submitting
[ ] feature request => Please check if request is not on the roadmap already https://github.com/primefaces/primeng/wiki/Roadmap
[ ] support request => Please do not submit support request here, instead see http://forum.primefaces.org/viewforum.php?f=35
```

**Plunkr Case (Bug Reports)**
Please demonstrate your case at stackblitz by using the issue template below. Issues without a test case have much less possibility to be reviewd in detail and assisted.

https://stackblitz.com/github/primefaces/primeng-issue-template

**Current behavior**

_Issue is visible in Showcase._
Open the ContextMenu example of the Table, right click on row, and then click away. The row remains selected.

**Expected behavior**

After clicking away from the context menu, the selection would be no longer valid.

**Minimal reproduction of the problem with instructions**

See Current behavior section.

**What is the motivation / use case for changing the behavior?**

**Comment by misterjaffacakes** [comment]
This is especially troublesome if your rows have click events attached to them and you want to know which is the last row you clicked on. For example:

You right click on a row to open the context menu, then you click on another row. Now you are left with two highlighted rows and confused about which one you clicked on. This is also the case in reverse order: you click on a row, then right click on another row; same result, two rows highlighted.

We had to add custom logic to remove the ui-contextmenu-selected and selected-highlight classes in order to avoid this issue. This also lends itself to other issues arising, where the respective class isn't added to the selected row and we needed to add it ourselves.
